Fix types in blocks/package/compponents

alexflorisca opened the pull request:
### Submission Review Guidelines:

This PR fixes all but 2 typescript errors in `client/blocks/packages/components`. The remaining 2 errors are related to the fact that we are referencing properties onloy available in the `@woocommerce/notices` package, but we are instead using the `@wordpress/notices` package. This should be fixed separately, as it will need thoughtful consideration around notices, out of scope for this PR.

Note I did not fix any errors in the test files

The issue is recorded in https://github.com/woocommerce/woocommerce/issues/56690

Closes https://github.com/woocommerce/woocommerce/issues/55975.


<!-- Begin testing instructions -->

### How to test the changes in this Pull Request:

This is only testable by developers and has no impact on the end user.

1. In your terminal `cd plugins/woocommerce/client/blocks`
2. `npx tsc --noEmit | grep -i "packages/components" | grep -iv "test"`
3. Check that there are only 2 type errors, related to the `explicitDismiss` property.
4. All Ci should pass

<!-- End testing instructions -->

### Testing that has already taken place:

<!-- Detail any testing that has already been conducted. -->
<!-- Include environment details such as hosting type, plugins, theme, store size, store age, and relevant settings. -->
<!-- Mention any analysis performed, such as assessing potential impacts on environment attributes and other plugins, performance profiling, or LLM/AI-based analysis. -->
<!-- Within the testing details you provide, please ensure that no sensitive information (such as API keys, passwords, user data, etc.) is included in this public pull request. -->

### Changelog entry

<!-- You can optionally choose to enter a changelog entry by checking the box below and supplying data. -->
<!-- It will trigger the 'Add changelog to PR' CI job to create and push the entry into the branch. -->

<!-- Due to org permissions, the job might fail for PRs crated from a fork under GitHub organizations. Possible solutions: -->
<!-- * Create entry manually with `pnpm --filter='@woocommerce/plugin-woocommerce' changelog add` and push it into the branch (replace `@woocommerce/plugin-woocommerce` with package name from nearest `package.json` file) -->
<!-- * Create entry from supplied PR data and push it automatically `pnpm utils changefile pr-number-here -o github-org-name-here` -->

-   [ ] Automatically create a changelog entry from the details below.

<!-- If no changelog entry is required for this PR, you can specify that below and provide a comment explaining why. This cannot be used if you selected the option to automatically create a changelog entry above. -->

-   [x] This Pull Request does not require a changelog entry. (Comment required below)

<details>

<summary>Changelog Entry Details</summary>

#### Significance

<!-- Choose only one -->

-   [ ] Patch
-   [ ] Minor
-   [ ] Major

#### Type

<!-- Choose only one -->

-   [ ] Fix - Fixes an existing bug
-   [ ] Add - Adds functionality
-   [ ] Update - Update existing functionality
-   [ ] Dev - Development related task
-   [ ] Tweak - A minor adjustment to the codebase
-   [ ] Performance - Address performance issues
-   [ ] Enhancement - Improvement to existing functionality

#### Message <!-- Add a changelog message here -->

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment <!-- If your Pull Request doesn't require a changelog entry, a comment explaining why is required instead -->
This PR only fixes some types and has no impact outside of developer experience.

</details>

Commit:
Fix CheckboxControl ts errors

Commit:
Fix ts errors in checkbox list options control

Commit:
Fix ts errors in chip component

Commit:
Fix rypes in panel and form-step components

Commit:
Fix Store Notices Container TS errors

Commit:
Change Notice Status type name

Commit:
Fix ts errors for FormattedMonetaryAmount

Commit:
Fix validation input error ts errors

Commit:
Remove SVG specific props

Commit:
Fix ts errors in storybook

wavvves reviewed (commented):
We have [JS unit tests failing](https://github.com/woocommerce/woocommerce/actions/runs/14094242257/job/39478128124?pr=56708) :/

Commit:
Set default chip element to li

Commit:
fix: Pass value prop to input element in CheckboxControl component

This fixes failing tests in the Rating Filter block where checkboxes were not being found by their value attribute because the value prop wasn't being passed through to the input element.

alexflorisca commented:
Thanks @wavvves, I missed that. All fixed, CI is passing. Let me know when you can do another review.

wavvves reviewed (approved):
Errors output seem ok, since this touched a lot of files and there's no testing instructions for functionality do you think another pair of eye in the code changes is needed?

mikejolley reviewed (approved):
Thank you @alexflorisca 🫡

I will merge this.
